instrumentation/aws-sdk/src/main/java/software/amazon/opentelemetry/javaagent/instrumentation/awssdk_v2_2/TracingExecutionInterceptor.java

```diff
@@ -0,0 +1,171 @@
+import software.amazon.awssdk.core.SdkRequest;
+import software.amazon.awssdk.core.SdkResponse;
+import software.amazon.awssdk.core.interceptor.Context;
+import software.amazon.awssdk.core.interceptor.ExecutionAttribute;
+import software.amazon.awssdk.core.interceptor.ExecutionAttributes;
+import software.amazon.awssdk.core.interceptor.ExecutionInterceptor;
+import software.amazon.awssdk.http.SdkHttpResponse;
+
+public class TracingExecutionInterceptor implements ExecutionInterceptor {
+  private static final String GEN_AI_SYSTEM_BEDROCK = "aws.bedrock";
+
+  private static final ExecutionAttribute<io.opentelemetry.context.Context> CONTEXT_ATTRIBUTE =
+      new ExecutionAttribute<>(TracingExecutionInterceptor.class.getName() + ".Context");
+
+  private static final ExecutionAttribute<Scope> SCOPE_ATTRIBUTE =
+      new ExecutionAttribute<>(TracingExecutionInterceptor.class.getName() + ".Scope");
+
+  private static final ExecutionAttribute<AwsSdkRequest> AWS_SDK_REQUEST_ATTRIBUTE =
+      new ExecutionAttribute<>(TracingExecutionInterceptor.class.getName() + ".AwsSdkRequest");
+
+  private final AwsSdkInstrumenterFactory instrumenterFactory; // ADDED
+
+  private static final Logger logger =
+      Logger.getLogger(TracingExecutionInterceptor.class.getName()); // ADDED
+
+  private static final ExecutionAttribute<RequestSpanFinisher> REQUEST_FINISHER_ATTRIBUTE =
+      new ExecutionAttribute<>(TracingExecutionInterceptor.class.getName() + ".RequestFinisher");
+
+  private final FieldMapper fieldMapper = new FieldMapper();
+
+  public TracingExecutionInterceptor() {
+    // for instantiation
+    this.instrumenterFactory = new AwsSdkInstrumenterFactory(GlobalOpenTelemetry.get());
+  }
+
+  @Override
+  public SdkRequest modifyRequest(
+      Context.ModifyRequest context, ExecutionAttributes executionAttributes) {
+    // This is the latest point where we can start the span, since we might need to inject
+    // it into the request payload. This means that HTTP attributes need to be captured later.
+    io.opentelemetry.context.Context parentOtelContext = io.opentelemetry.context.Context.current();
+    SdkRequest request = context.request();
+
+    // Ignore presign request. These requests don't run all interceptor methods and the span
+    // created
+    // here would never be ended and scope closed.
+    if (executionAttributes.getAttribute(AwsSignerExecutionAttribute.PRESIGNER_EXPIRATION)
+        != null) {
+      return request;
+    }
+
+    Instrumenter<ExecutionAttributes, Response> instrumenter =
+        instrumenterFactory.requestInstrumenter();
+
+    if (!instrumenter.shouldStart(parentOtelContext, executionAttributes)) {
+      // NB: We also skip injection in case we don't start.
+      return request;
+    }
+
+    RequestSpanFinisher requestFinisher = instrumenter::end;
+    io.opentelemetry.context.Context otelContext =
+        instrumenter.start(parentOtelContext, executionAttributes);
```

Comment:
I think this will start a new span. I am not sure why we want to do it because we are only supposed to modify attributes of the existing spans generated by upstream aws instrumentation.